### DISCUSSION TITLE
Teleport HA and session recordings

### DISCUSSION THREAD

**Discussion by hydezhao** [gravitational/teleport#19236] [Q&A] [answered]
Hello,

I am setting up a Teleport HA cluster with 2 auth servers behind a LB and 2 proxy servers behind a LB, auth servers are connecting to a etcd cluster backend for core state data, and session recordings are configured to sent to a S3 bucket. 

While I found that all session recordings are not always visible on Web UI or `tsh recordings ls`. When refreshing the Web page or try the CLI commands multiple times, there is only a part of session recordings is available, and it changes with refreshing. 

Is it normal or I missed some configs ?

Thanks in advance.

**Comment by webvictim** [comment, marked as answer]
This is unfortunately normal and expected behaviour as there is no supported on-premise HA backend. The only supported options for HA audit log storage are AWS DynamoDB and Google Firestore.

The reason the error happens is because each time your proxy connects to a different auth server, it sees a different list of audit logs (which is used to construct the view of session recordings) based on the audit log files, which live on each auth server's local storage.

The best solution we can offer you is to run something (like Filebeat or [Teleport's event handler plugin](https://goteleport.com/docs/management/guides/fluentd/)) to collect the audit log files on all auth servers and forward them to a SIEM where you can then get a unified view of all audit logs across your cluster. The web UI views of audit logs and session recordings will always be broken in this case, though.

Sorry :(

**Reply by hydezhao** [reply]
Thank you @webvictim for your reply. It is good ( and sad ) to know that proxy uses the audit logs to construct the view of session recordings.. which means if the audit logs are rotated, even if the session recording are stored in s3 , we cannot see it in WebUI or tsh cli anymore. :( 

If the session recordings part of information could be stored along with core state ( in etcd ) , we won't have this trouble, ( yes I know it is always easier to say than to dev ). 

Btw, do you think, if I mount a NFS between two auth servers for audit logs storage, it will work ?

**Reply by webvictim** [reply]
Apparently `etcd` isn't well suited for highly active time series data like audit logs which is why it hasn't been implemented already.

You can try NFS but you may find that you run into weird file locking issues. We don't recommend it.

**Reply by hydezhao** [reply]
I raised a feature request to try my chance : https://github.com/gravitational/teleport/issues/19269
